lib/valueflow.cpp

```diff
@@ -1551,27 +1589,6 @@ static bool valueFlowForward(Token * const               startToken,
                 }
             }
 
-            // bailout if address of var is taken..
```

Comment:
I wrote:
> If for some reason the original order is needed then it must be checked in CheckMove::checkAccess() if the moved variable is an (in-)out parameter of a function. This is possible, too.

The function valueFlowSubFunction does make use of the value of the argument. Even though it currently does not do it for non-const reference parameters since value flow bails out in this case in 
```
static void valueFlowInjectParameter(TokenList* tokenlist, ErrorLogger* errorLogger, const Settings* settings, const Variable* arg, const Scope* functionScope, const std::list<ValueFlow::Value>& argvalues)
{
    // Is argument passed by value or const reference, and is it a known non-class type?
    if (arg->isReference() && !arg->isConst() && !arg->isClass())
        return;
```

But I do not want to block an improvement here.
Therefore, I will keep the original order of the code in lib/valueflow.cpp and instead add a check in ```CheckMove::checkAccess()``` if the moved variable is an (in-)out parameter of a function.

PS: Maybe PKEuS can give some insights why valueFlowInjectParameter bails out for non-const reference parameters.